### DISCUSSION TITLE
SBOM related question

### DISCUSSION THREAD

**Discussion by harshitasao** [kubescape/kubescape#829] [Q&A]
Hey @dwertent @slashben, I was working on issue #668 and needed to generate the SBOM, so I did some research on Google and found some tools like [syft](https://github.com/anchore/syft) and [bom](https://github.com/kubernetes-sigs/bom) that could help. Please correct me if I'm not going in the right direction.

**Comment by slashben** [comment]
We already use grype (which is the sibling project of syft) for finding vulnerabilities. 

See this comment
https://github.com/kubescape/kubescape/issues/668#issuecomment-1244946591

**Reply by harshitasao** [reply]
Thanks for the quick response, @slashben. As you mentioned, I saw the comment, but from what I understand, the [README](https://github.com/anchore/grype#:~:text=Grype%20supports%20input%20of%20Syft%2C%20SPDX%2C%20and%20CycloneDX%20SBOM%20formats.%20If%20Syft%20has%20generated%20any%20of%20these%20file%20types%2C%20they%20should%20have%20the%20appropriate%20information%20to%20work%20properly%20with%20Grype.%20It%20is%20also%20possible%20to%20use%20SBOMs%20generated%20by%20other%20tools%20with%20varying%20degrees%20of%20success) of grype says that it takes the generated SBOM from syft or other tools and does the scanning. So, in order to generate the SBOM, we will need to use syft or another tool.

**Comment by slashben** [comment]
Grype does not need SBOM, it can create it byself

**Reply by harshitasao** [reply]
Thank you for clarifying, @slashben. Additionally, I would like to submit a proposal for this issue as part of the Open source Fest, so could you kindly share the proposal format to be used and the deadline for submission.

**Reply by slashben** [reply]
I suggest you open an issue and write your proposal there. In this way we can comment :)

**Comment by jaiprasad266** [comment]
just want to share my thought here , if we use the SBOM generated using syft and then use the SBOM as input to grype , grype will look for vulnerabilities and give out the report , the whole process takes under a second , where as if u use just grype without SBOM it will take around 10 to 15 seconds to do the same task .

**Reply by dwertent** [reply]
We were thinking of using Grype for generating the SBOM as well.

The flow will be as follows ->
```
Scan image:
    SBOM not found:
        Generate SBOM
     Scan SBOM
```
